src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,118 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData = new Object [DEFAULT_CAPACITY];
+    private int size = 0;
+
     @Override
     public void add(T value) {
 
+        if (size == elementData.length) {
+            grow(size);
+        }
+        elementData[size] = value;
+        size++;
+
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, 0, size);
+        if (size == elementData.length) {
+            grow(size);
+        }
+
+        for (int i = size; i > index; i--) {
+            elementData[i] = elementData[i - 1];
+        }
```

Comment:
lets use System.arraycopy() (this method can move elements within the same array